libsolidity/parsing/Token.h

```diff
@@ -394,14 +302,43 @@ class Token
 		return m_precedence[tok];
 	}
 
-	static Token::Value fromIdentifierOrKeyword(std::string const& _name);
+	static std::tuple<Token::Value, unsigned short, unsigned short> fromIdentifierOrKeyword(std::string const& _literal);
 
 private:
+	// extractM and extractMxN provide a safe way to extract numbers, 
+	// if out_of_range error is thrown, they returns 0s, therefore securing 
+	// the variable's identity as an identifier.
+	static unsigned extractM(std::string const& _literal);
+	static std::pair<unsigned, unsigned> extractMxN(std::string const& _literal);
 	static char const* const m_name[NUM_TOKENS];
 	static char const* const m_string[NUM_TOKENS];
 	static int8_t const m_precedence[NUM_TOKENS];
 	static char const m_tokenType[NUM_TOKENS];
 };
 
+class ElementaryTypeNameToken
+{
+public:
+	ElementaryTypeNameToken(Token::Value _token, unsigned const& _firstNumber, unsigned const& _secondNumber)
+	{
+		parseDetails(_token, _firstNumber, _secondNumber);
+		m_token = _token;
+	}
+
+	unsigned int const& firstNumber() const { return m_firstNumber; }
+	unsigned int const& secondNumber() const { return m_secondNumber; }
+	Token::Value const& returnTok() const { return m_token; }
```

Comment:
Please do not name this `returnTok` - this style is not used anywhere else. Please use `token` oh and no need to return integers by `const&`.